tests/test_dependencies.py

```diff
@@ -54,3 +54,84 @@ def test_transitive_v2_dependencies(examples_v2_package_index, pytestconfig, tes
             result.assert_outcomes(passed=1)
         finally:
             utils.unload_modules_for_path(venv.site_packages_dir)
+
+
+def test_conflicing_dependencies_strict(
+    examples_v2_package_index, pytestconfig, testdir
+):
+    # set working directory to allow in-place with all example modules
+    pytest_inmanta.plugin.CURDIR = str(
+        pytestconfig.rootpath / "examples" / "test_conflict_dependencies"
+    )
+    """
+    when using the pytest-inmanta without specifying the --no-strict-deps-check, the constraints
+    of the installed modules/packages are veryfied and if a conflict is detected a ConflictingRequirement
+    error is raised
+    """
+    testdir.copy_example("test_conflict_dependencies")
+
+    with tempfile.TemporaryDirectory() as venv_dir:
+        # set up environment
+        venv: env.VirtualEnv = env.VirtualEnv(env_path=venv_dir)
+        try:
+            venv.use_virtual_env()
+
+            # run tests
+            result = testdir.runpytest_inprocess(
+                "tests/test_basics.py",
+                "--use-module-in-place",
+                # add pip index containing examples packages as module repo
+                "--module_repo",
+                f"package:{examples_v2_package_index}",
+                # include configured pip index for inmanta-module-std
+                "--module_repo",
+                "package:"
+                + os.environ.get("PIP_INDEX_URL", "package:https://pypi.org/simple"),
+            )
+            result.assert_outcomes(errors=1)
+            assert "ConflictingRequirements" "\n".join(result.outlines)
+        finally:
+            utils.unload_modules_for_path(venv.site_packages_dir)
+
+
+def test_conflicing_dependencies_no_strict(
```

Comment:
Why not just parametrize the test case? As far as I see there's only one word difference.